app/components/form/fields/ImageSelectField.tsx

```diff
@@ -44,20 +44,48 @@ export function ImageSelectField({ images, control }: ImageSelectFieldProps) {
 const Slash = () => <span className="mx-0.5 text-quinary">/</span>
 
 export function toListboxItem(i: Image, includeProjectSiloIndicator = false): ListboxItem {
-  const projectSiloIndicator = includeProjectSiloIndicator ? (
+  const { name, os, projectId, size, version } = i
+  const formattedSize = `${bytesToGiB(size, 1)} GiB`
+
+  // filter out any undefined metadata and create a comma-separated list
+  // for the selected listbox item (shown in labelString)
+  const condensedImageMetadata = [os, version, formattedSize].filter((i) => !!i).join(', ')
+  const metadataForLabelString = condensedImageMetadata.length
+    ? ` (${condensedImageMetadata})`
+    : ''
+
+  // for metadata showing in the dropdown's options, include the project / silo indicator if requested
+  const versionInfo = includeProjectSiloIndicator ? (
     <>
-      <Slash /> {i.projectId ? 'Project image' : 'Silo image'}
+      {version} <Slash /> {projectId ? 'Project image' : 'Silo image'}
     </>
-  ) : null
+  ) : (
+    version
+  )
+  // filter out undefined metadata here, as well, and create a `<Slash />`-separated list
+  // for the listbox item (shown for each item in the dropdown)
+  const metadataForLabel = [os, versionInfo, formattedSize]
+    .filter((i) => !!i)
+    .map((i, index) => (
+      <span key={`${i}`}>
+        {index > 0 ? (
+          <>
+            {' '}
+            <Slash />{' '}
+          </>
+        ) : (
+          ''
+        )}
+        {i}
+      </span>
+    ))
```

Comment:
You can avoid this evil JSX formatting situation by increasing the margin around the slash and getting rid of the spaces around the other usage:

```diff
diff --git a/app/components/form/fields/ImageSelectField.tsx b/app/components/form/fields/ImageSelectField.tsx
index 2ba69fdf..256fd390 100644
--- a/app/components/form/fields/ImageSelectField.tsx
+++ b/app/components/form/fields/ImageSelectField.tsx
@@ -41,7 +41,7 @@ export function ImageSelectField({ images, control }: ImageSelectFieldProps) {
   )
 }
 
-const Slash = () => <span className="mx-0.5 text-quinary">/</span>
+const Slash = () => <span className="mx-1 text-quinary">/</span>
 
 export function toListboxItem(i: Image, includeProjectSiloIndicator = false): ListboxItem {
   const { name, os, projectId, size, version } = i
@@ -57,7 +57,9 @@ export function toListboxItem(i: Image, includeProjectSiloIndicator = false): Li
   // for metadata showing in the dropdown's options, include the project / silo indicator if requested
   const versionInfo = includeProjectSiloIndicator ? (
     <>
-      {version} <Slash /> {projectId ? 'Project image' : 'Silo image'}
+      {version}
+      <Slash />
+      {projectId ? 'Project image' : 'Silo image'}
     </>
   ) : (
     version
@@ -68,14 +70,7 @@ export function toListboxItem(i: Image, includeProjectSiloIndicator = false): Li
     .filter((i) => !!i)
     .map((i, index) => (
       <span key={`${i}`}>
-        {index > 0 ? (
-          <>
-            {' '}
-            <Slash />{' '}
-          </>
-        ) : (
-          ''
-        )}
+        {index > 0 ? <Slash /> : ''}
         {i}
       </span>
     ))
```

Reply:
Actually... I think `Project image`/`Silo image` label is supposed to go on the end. That makes it even better:

```diff
diff --git a/app/components/form/fields/ImageSelectField.tsx b/app/components/form/fields/ImageSelectField.tsx
index 2ba69fdf..73cf9c11 100644
--- a/app/components/form/fields/ImageSelectField.tsx
+++ b/app/components/form/fields/ImageSelectField.tsx
@@ -41,7 +41,7 @@ export function ImageSelectField({ images, control }: ImageSelectFieldProps) {
   )
 }
 
-const Slash = () => <span className="mx-0.5 text-quinary">/</span>
+const Slash = () => <span className="mx-1 text-quinary">/</span>
 
 export function toListboxItem(i: Image, includeProjectSiloIndicator = false): ListboxItem {
   const { name, os, projectId, size, version } = i
@@ -55,27 +55,19 @@ export function toListboxItem(i: Image, includeProjectSiloIndicator = false): Li
     : ''
 
   // for metadata showing in the dropdown's options, include the project / silo indicator if requested
-  const versionInfo = includeProjectSiloIndicator ? (
-    <>
-      {version} <Slash /> {projectId ? 'Project image' : 'Silo image'}
-    </>
-  ) : (
-    version
-  )
+  const projectSiloIndicator = includeProjectSiloIndicator
+    ? projectId
+      ? 'Project image'
+      : 'Silo image'
+    : null
+
   // filter out undefined metadata here, as well, and create a `<Slash />`-separated list
   // for the listbox item (shown for each item in the dropdown)
-  const metadataForLabel = [os, versionInfo, formattedSize]
+  const metadataForLabel = [os, version, formattedSize, projectSiloIndicator]
     .filter((i) => !!i)
     .map((i, index) => (
       <span key={`${i}`}>
-        {index > 0 ? (
-          <>
-            {' '}
-            <Slash />{' '}
-          </>
-        ) : (
-          ''
-        )}
+        {index > 0 ? <Slash /> : ''}
         {i}
       </span>
     ))
```